src/ui/NoNameFacet/NoNameFacetHeader.ts

```diff
@@ -88,23 +92,66 @@ export class NoNameFacetHeader {
   }
 
   private recreateOperator() {
-    $$(this.operatorElement).replaceWith(this.createOperator());
+    const newElement = this.createOperator();
+    $$(this.operatorElement).replaceWith(newElement);
+    this.operatorElement = newElement;
+  }
+
+  private createCollapse() {
+    const { enableTogglingCollapse, isCollapsed } = this.options.rootFacetOptions;
+    const svgIcon = isCollapsed ? SVGIcons.icons.facetExpand : SVGIcons.icons.facetCollapse;
+    const collapseBtn = $$('button', { className: 'coveo-facet-header-collapse' }, svgIcon);
+    const className = `coveo-facet-settings-section-${isCollapsed ? 'show' : 'hide'}-svg`;
+    SVGDom.addClassToSVGInContainer(collapseBtn.el, className);
+
+    const label = isCollapsed ? l('Expand') : l('Collapse');
+    collapseBtn.setAttribute('aria-label', label);
+    collapseBtn.setAttribute('title', label);
+
+    collapseBtn.on('click', this.toggleCollapseAction);
+    collapseBtn.toggle(enableTogglingCollapse);
+
+    return collapseBtn.el;
+  }
+
+  private recreateCollapse() {
+    const newElement = this.createCollapse();
+    $$(this.collapseElement).replaceWith(newElement);
+    this.collapseElement = newElement;
   }
 
-  private clearAction() {}
+  private clearAction = () => {};
+
+  private toggleOperatorAction = () => {
+    this.options.rootFacetOptions.useAnd = !this.options.rootFacetOptions.useAnd;
+    this.recreateOperator();
+  };
 
-  private toggleOperatorAction() {}
+  private toggleCollapseAction = () => {
```

Comment:
What about `toggleCollapse` for the name, or even `toggle` if this button was in it's own file as per the comment above.